hsds/app.py

```diff
@@ -329,6 +331,27 @@ def main():
     if args.dn_count:
         kwargs["dn_count"] = args.dn_count
 
+    if args.bucket_name:
+        bucket_name = args.bucket_name
+    else:
+        bucket_name = config.get("bucket_name")
+    if not bucket_name:
+        sys.exit("bucket_name not set")
+    if args.root_dir:
+        root_dir = args.root_dir
+    else:
+        root_dir = config.get("root_dir")
+    if not root_dir:
+        # check that AWS_S3_GATEWAY or AZURE_CONNECTION_STRING is set
+        if not config.get("aws_s3_gateway") and not config.get("azure_connection_string"):
+            sys.exit("root_dir not set (and no S3 or Azure connection info)")
+    else:
+        if not os.path.isdir(root_dir):
+            sys.exit(f"directory: {root_dir} not found")
+        bucket_path = os.path.join(root_dir, bucket_name)
```

Comment:
@jreadey I think there is an issue here.  The bucket name retrieved from the arg parser is in the form of a single-element list due to specifying [nargs=1](https://docs.python.org/3/library/argparse.html#nargs). This call to `os.path.join` then errors when passed a list.